salt/modules/syslog_ng.py

```diff
@@ -784,49 +784,27 @@ def set_parameters(version=None,
     return _format_return_data(0)
 
 
-def _add_to_path_envvar(directory):
-    '''
-    Adds directory to the PATH environment variable and returns the original
-    one.
-    '''
-    orig_path = os.environ.get('PATH', '')
-    if directory:
-        if not os.path.isdir(directory):
-            log.error('The given parameter is not a directory')
-
-        os.environ['PATH'] = '{0}{1}{2}'.format(orig_path,
-                                                os.pathsep,
-                                                directory)
-    return orig_path
-
-
-def _restore_path_envvar(original):
-    '''
-    Sets the PATH environment variable to the parameter.
-    '''
-    if original:
-        os.environ['PATH'] = original
-
-
 def _run_command_in_extended_path(syslog_ng_sbin_dir, command, params):
     '''
-    Runs the given command in an environment, where the syslog_ng_sbin_dir is
-    added then removed from the PATH.
+    Runs the specified command with the syslog_ng_sbin_dir in the PATH
     '''
-    orig_path = _add_to_path_envvar(syslog_ng_sbin_dir)
-
-    if not salt.utils.path.which(command):
-        error_message = (
-            'Unable to execute the command \'{0}\'. It is not in the PATH.'
-            .format(command)
-        )
-        log.error(error_message)
-        _restore_path_envvar(orig_path)
-        raise CommandExecutionError(error_message)
-
-    ret = _run_command(command, options=params)
-    _restore_path_envvar(orig_path)
-    return ret
+    orig_path = os.environ.get('PATH', '')
+    env = None
+    if syslog_ng_sbin_dir:
+        if not os.path.isdir(syslog_ng_sbin_dir):
+            log.error('syslog_ng_sbin_dir path %s is not a directory',
+                      syslog_ng_sbin_dir)
+        else:
+            env = {
+                str('PATH'): salt.utils.stringutils.to_str(  # future lint: disable=blacklisted-function
```

Comment:
A comment about this `str()` and `to_str()` behavior would likely be good here as well.